src/plugins/core/conditional_format.ts

```diff
@@ -58,37 +58,99 @@ export class ConditionalFormatPlugin
 
   readonly cfRules: { [sheet: string]: ConditionalFormatInternal[] } = {};
 
-  loopThroughRangesOfSheet(sheetId: UID, applyChange: ApplyRangeChange) {
-    for (const rule of this.cfRules[sheetId]) {
-      if (rule.rule.type === "DataBarRule" && rule.rule.rangeValues) {
-        const change = applyChange(rule.rule.rangeValues);
-        switch (change.changeType) {
-          case "REMOVE":
-            this.history.update(
-              "cfRules",
-              sheetId,
-              this.cfRules[sheetId].indexOf(rule),
-              "rule",
-              //@ts-expect-error
-              "rangeValues",
-              undefined
-            );
-            break;
-          case "RESIZE":
-          case "MOVE":
-          case "CHANGE":
+  loopThroughAllRanges(applyChange: ApplyRangeChange) {
+    for (const sheetId of Object.keys(this.cfRules)) {
+      for (const rule of this.cfRules[sheetId]) {
+        if (rule.rule.type === "DataBarRule" && rule.rule.rangeValues) {
+          const change = applyChange(rule.rule.rangeValues);
+          switch (change.changeType) {
+            case "REMOVE":
+              this.history.update(
+                "cfRules",
+                sheetId,
+                this.cfRules[sheetId].indexOf(rule),
+                "rule",
+                //@ts-expect-error
+                "rangeValues",
+                undefined
+              );
+              break;
+            case "RESIZE":
+            case "MOVE":
+            case "CHANGE":
+              this.history.update(
+                "cfRules",
+                sheetId,
+                this.cfRules[sheetId].indexOf(rule),
+                "rule",
+                //@ts-expect-error
+                "rangeValues",
+                change.range
+              );
+              break;
+          }
+        } else if (rule.rule.type === "CellIsRule") {
+          for (let arg = 0; arg < rule.rule.values.length; arg++) {
```

Comment:
`arg` ? Not sure what is means here ?
```suggestion
          for (let i = 0; i < rule.rule.values.length; i++) {
```